src/DataLoader.Abstractions/IDataLoaderContext.cs

```diff
@@ -5,17 +5,28 @@
 namespace Chinook.DataLoader
 {
 	/// <summary>
-	/// A <see cref="IDataLoaderContext"/> represents metadata that
-	/// was added to the <see cref="IDataLoaderRequest"/> before and after its execution.
+	/// A <see cref="IDataLoaderContext"/> represents metadata added to the <see cref="IDataLoaderRequest"/> before and during its execution.
 	/// </summary>
-	public interface IDataLoaderContext
+	public interface IDataLoaderContext : IDictionary<string, object>
 	{
 		/// <summary>
 		/// Gets or sets a value for the given key.
+		/// If the key is not found, null is returned.
 		/// </summary>
 		/// <param name="key">Key</param>
-		/// <remarks>If the key is not found, null is returned.</remarks>
-		/// <returns>Value</returns>
-		object this[string key] { get; set; }
+		/// <returns>The value associated to the key when available; null otherwise.</returns>
+		new object this[string key] { get; set; }
```

Comment:
Because it's the same signature, but different behavior and documentation. When you use the indexer on `IDataLoaderContext`, you want IntelliSense to show you this xml doc, not the one from `IDictionary`.